webapp/shop/cred/views.py

```diff
@@ -206,7 +213,17 @@ def cred_your_exams(ua_contracts_api, trueability_api, **_):
                         "button_class": "p-button",
                     }
                 ]
-                exams.append({"name": name, "actions": actions})
+                exams.append(
+                    {"name": name, "state": "Not taken", "actions": actions}
+                )
+
+    exams = (
+        [exam for exam in exams if exam["state"] == "In progress"]
+        + [exam for exam in exams if exam["state"] == "Scheduled"]
+        + [exam for exam in exams if exam["state"] == "Not taken"]
+        + [exam for exam in exams if exam["state"] == "Complete"]
+        + [exam for exam in exams if exam["state"] == "Cancelled"]
+    )
```

Comment:
Open to suggestions if there is a cleaner way to filter and sort by the exam state.

Reply:
Instead of `exams = []` declare the following at the top of your function.
```
exams_in_progress = []
exams_scheduled = []
exams_not_taken = []
exams_complete = []
exams_cancelled = []
```
As you go through the exams append to the relevant list instead of `exams` and then do:
`exams = exams_in_progress + exams_scheduled + exams_not_taken + exams_complete + exams_cancelled`

Reply:
I agree with Joao. Group them in separate arrays when you go through the exams.